test/distributed/test_c10d.py

```diff
@@ -3056,11 +3056,11 @@ def allreduce_with_then_hook(
 
             def mult(fut):
                 # Multiply the result by 2.
-                return [2 * t for t in fut.wait()]
+                return [2 * t for t in fut.value()]
```

Comment:
Shall we also add a `value()` wrapper method to `torch.futures.Future` and add docs there?